lib/cosmos-db/cosmos-db.providers.ts

```diff
@@ -52,9 +52,17 @@ export function createAzureCosmosDbProviders(
       }
 
       if (partitionKey != null) {
-        containerOptions.partitionKey = {
-          paths: [`/${partitionKey}`],
-        };
+        if (typeof partitionKey === 'string') {
+          containerOptions.partitionKey = {
+            paths: [`/${partitionKey}`],
+          };
+        }
+        else if (typeof partitionKey === 'object') {
```

Comment:
I think the partition key might be an array, which will still resolve to object and in this case will work just fine. Looks good to me :)